src/search/search-index/search.js

```diff
@@ -252,12 +301,18 @@ export async function search(
     console.time('filter search')
     const filterPageResultsMap = await filterSearch(query)
     console.timeEnd('filter search')
+    console.time('bookmark search')
+    const bookMarksResultMap = query.bookmarksFilter
+        ? await bookmarkFilterBackSearch(query)
+        : null
```

Comment:
Instead of defaulting to `null` here, I think better to put a check at the start of `bookmarkFilterBackSearch` which returns `null` if `bookmarksFilter` is `false` (available in the query object)